include/pmacc/algorithms/math/PowerFunction.hpp

```diff
@@ -0,0 +1,64 @@
+/* Copyright 2022 Brian Marre, Rene Widera, Richard Pausch
+ *
+ * This file is part of PMacc.
+ *
+ * PMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with PMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#ifndef unitTest
+#    define unitTest true
+#endif
+
+#include "pmacc/types.hpp"
+#if unitTest == true
+#    include "pmacc/static_assert.hpp"
+#endif
+
+#include <cstdint>
+
+namespace pmacc::math
+{
+    /** power function for integer exponents, constexpr
+     *
+     * @tparam T_Type return and accumulation data type
+     * @tparam T_Exp exponent data type, default uint32_t
+     *
+     * @param x base
+     * @param exp exponent
+     */
+    template<typename T_Type, typename T_Exp = uint32_t>
+    HDINLINE constexpr T_Type pow(T_Type const x, T_Exp const exp)
+    {
+        T_Type result = static_cast<T_Type>(1u);
+        for(T_Exp e = static_cast<T_Exp>(0u); e < exp; e++)
```

Comment:
@psychocoderHPC the actual reason for using template meta programming was to unroll this for loop.
Can we make sure that this for loop is unrolled on any architecture with today's compilers.

Reply:
added a PMACC_UNROLL(100), should be plenty for our us

Reply:
Nevertheless a modern C++14-Compiler should unroll and evaluate at compile time, we can check using register usage and field solver test for the AOFD-solver, since it uses it in a fieldSolver-hot loop.
@psychocoderHPC agreed to take a look.
Add. Offline discussion: Probably it is not that important that the power function is evaluated at compile time, but the AOFD-weight array